pkg/webui/console/views/device-general-settings/index.js

```diff
@@ -207,7 +223,7 @@ export default class DeviceGeneralSettings extends React.Component {
     // 3. Disable the seciton if `join_server_address` is not equal to the cluster JS address.
     // 4. Disable the seciton if an external JS is used.
     const sameJsAddress = getComponentBaseUrl(jsConfig) === device.join_server_address
-    const externalJs = hasExternalJs(device)
+    const externalJs = hasExternalJs(device) && mayReadKeys
     const jsDisabled = !jsEnabled || !isOTAA || !sameJsAddress || externalJs
```

Comment:
I think it's confusing if we disable the entire JS section only because someone isn't allowed to read keys.